dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ResourceLinkResource.java

```diff
@@ -0,0 +1,150 @@
+    @JSONP
+    @NoCache
+    @Produces(MediaType.APPLICATION_JSON + ";charset=UTF-8")
+    public Response findResourceLink(@Context final HttpServletRequest  request,
+                                     @Context final HttpServletResponse response,
+                                     @QueryParam("inode")            final String inode,
+                                     @QueryParam("identifier")       final String identifier,
+                                     @DefaultValue("-1") @QueryParam("language") final String language) throws DotStateException, DotSecurityException, DotDataException {
+        if (!UtilMethods.isSet(inode) && !UtilMethods.isSet(identifier)) {
+
+            throw new IllegalArgumentException("Missing required inode/identifier param");
+        }
+
+        final InitDataObject initData =
+                new WebResource.InitBuilder(webResource).requiredBackendUser(true).requiredFrontendUser(false)
+                        .requestAndResponse(request, response).rejectWhenNoUser(true).init();
+        final User user       = initData.getUser();
+        final long languageId = LanguageUtil.getLanguageId(language);
+        final PageMode mode   = PageMode.get(request);
+        final Contentlet contentlet = this.getContentlet(inode, identifier, languageId,
+                ()-> WebAPILocator.getLanguageWebAPI().getLanguage(request).getId(), initData, mode);
+
+        Logger.debug(this, ()-> "Finding the resource link for the contentlet: " + contentlet.getIdentifier());
+        final ResourceLink link     = new ResourceLink.ResourceLinkBuilder().build(request, user, contentlet);
+        if(link.isDownloadRestricted()) {
+
+            throw new DotSecurityException("The Resource link to the contentlet is restricted.");
+        }
+
+        return Response.ok(new ResponseEntityView(ImmutableMap.of("resourceLink",
+                ImmutableMap.of(
+                        "href",        link.getResourceLinkAsString(),
+                        "text",        link.getResourceLinkUriAsString(),
+                        "mimeType",    link.getMimeType(),
+                        "idPath",      link.getIdPath(),
+                        "versionPath", link.getVersionPath()
+                )
+        ))).build();
+
+    }
+
+    private Contentlet getContentlet(final String inode,
+                                     final String identifier,
+                                     final long language,
+                                     final Supplier<Long> sessionLanguage,
+                                     final InitDataObject initDataObject,
+                                     final PageMode pageMode) throws DotDataException, DotSecurityException {
+
+        Contentlet contentlet = null;
+        PageMode mode         = pageMode;
+
+        if(UtilMethods.isSet(inode)) {
+
+            Logger.debug(this, ()-> "Finding the contentlet by inode: " + inode);
+            contentlet = this.contentletAPI.find
```

Comment:
why do it not using the language here?

Reply:
b/c the inode implicit includes the language, inode = id + lang (in some version)